internal/service/vpc/resource_aws_vpc_peering_connection.go

```diff
@@ -0,0 +1,337 @@
+	}
+
+	diags := diag.Diagnostics{}
+	pc = res.(*aiven.VPCPeeringConnection)
+	if pc.State != "ACTIVE" {
+		switch pc.State {
+		case "PENDING_PEER":
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary: fmt.Sprintf("Aiven platform has created a connection to the specified "+
+					"peer successfully in the cloud, but the connection is not active until the user "+
+					"completes the setup in their cloud account. The steps needed in the user cloud "+
+					"account depend on the used cloud provider. Find more in the state info: %s",
+					stateInfoToString(pc.StateInfo)),
+			})
+		case "DELETED":
+			diags = append(diags, diag.Errorf("A user has deleted the peering connection through the Aiven "+
+				"Terraform provider, or Aiven Web Console or directly via Aiven API. There are no "+
+				"transitions from this state")...)
+		case "DELETED_BY_PEER":
+			diags = append(diags, diag.Errorf("A user deleted the peering cloud resource in their account. "+
+				"There are no transitions from this state")...)
+		case "REJECTED_BY_PEER":
+			diags = append(diags, diag.Errorf("AWS VPC peering connection request was rejected, state info: %s",
+				stateInfoToString(pc.StateInfo))...)
+		case "INVALID_SPECIFICATION":
+			diags = append(diags, diag.Errorf("VPC peering connection cannot be created, more in the state info: %s",
+				stateInfoToString(pc.StateInfo))...)
+		default:
+			return diag.Errorf("Unknown VPC peering connection cannot state: %s", pc.State)
+		}
+	}
+
+	d.SetId(schemautil.BuildResourceID(projectName, vpcID, pc.PeerCloudAccount, pc.PeerVPC, *pc.PeerRegion))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		return append(diags, resourceAWSVPCPeeringConnectionDelete(ctx, d, m)...)
+	}
+
+	return append(diags, resourceAWSVPCPeeringConnectionRead(ctx, d, m)...)
+}
+
+func resourceAWSVPCPeeringConnectionRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	pc, err := client.VPCPeeringConnections.GetVPCPeering(
+		projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+	}
+
+	return copyAWSVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
+}
+
+func resourceAWSVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+
+	err := client.VPCPeeringConnections.DeleteVPCPeering(
+		projectName,
+		vpcID,
+		peerCloudAccount,
+		peerVPC,
+		peerRegion,
+	)
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.Errorf("Error deleting VPC peering connection: %s", err)
+	}
+
+	stateChangeConf := &resource.StateChangeConf{
```

Comment:
Looks like this state change configuration is almost the same as in `resourceAWSVPCPeeringConnectionCreate()` except `Pending` and `Target` states. Can it be extracted to a function?